\section{Comparative Analysis with Baseline Methods}
To evaluate HYDRA's performance relative to the current state-of-the-art, we compared it against RFDiffusion \cite{watson2023novo}. RFDiffusion leverages a diffusion model framework to iteratively refine candidate peptides for target receptor binding. It begins with a pool of random sequences and progressively adjusts them based on predicted binding scores, aiming to converge toward high-affinity binders over multiple iterations \cite{watson2023novo}. While RFDiffusion stopped at predicting just the peptide backbone, we went a step further by utilizing ProteinMPNN \cite{dauparas2022robust} to generate full peptide sequences from these backbones for analysis of physicochemical properties. \\

\section{Case Study 1: PepBDB Protein Targets}
We generated 10 peptide binders for each binding pocket using RFDiffusion, mirroring the sampling strategy used with HYDRA. Since RFDiffusion solely predicts backbones, we used ProteinMPNN to translate each backbone into 3 candidate peptide sequences. The residue lengths for both RFDiffusion and ProteinMPNN were determined programmatically using the same algorithm employed by HYDRA. Finally, all generated peptides (from both HYDRA and RFDiffusion) underwent identical analyses, encompassing statistical properties, physicochemical properties, and binding affinity predictions. \\

Our evaluation demonstrates that HYDRA-designed peptides exhibit significantly better binding affinities to receptor pockets compared to those generated by RFDiffusion. As shown in Figure \ref{fig:performance}A, for approximately 65\% of the target receptors, HYDRA-generated peptides achieved higher mean FRODOCK scores. This is further corroborated by Figure \ref{fig:performance}B, where HYDRA peptides display lower mean binding affinity scores (indicating stronger binding) compared to RFDiffusion for roughly 78\% of the targets. This suggests that complexes formed by target receptors and HYDRA-designed peptides possess significantly greater stability. 
Additionally, we computed evaluation metrics using median and Top1 aggregations. Due to the nature of binding affinity predictions, when evaluating a set of peptide metrics, the median can provide a more reliable representation of the central tendency of the data, especially if there are a few peptides with exceptionally high or low values. The Top1 metric focuses on the single peptide with the best evaluation score and highlights the model's capability to identify the absolute best candidate peptide within a set. Figures \ref{fig:extended_vina_metrics} and \ref{fig:extended_frodock_metrics} illustrate the complete distributions of different aggregations of binding affinity metrics. \\

\begin{figure}
  \center

  \textbf{\Large A}
  \psfig{figure=figures/plots/gen_frodock_common_mean.pdf,width=\linewidth} \\

  \textbf{\Large B}
  \psfig{figure=figures/plots/gen_vina_common_mean.pdf,width=\linewidth}

  \textbf{\Large C}
  \psfig{figure=figures/plots/gen_diversity.pdf,width=\linewidth}

  \caption{\textbf{Comparison of binding affinity and peptide diversity between HYDRA and RFDiffusion.} Figures \textbf{(A)} and \textbf{(B)} depict the mean binding affinities and FRODOCK correlation scores, respectively, for peptides generated by each model. Lower binding affinities and higher FRODOCK scores indicate stronger predicted protein-peptide complexes. Binding targets are sorted based on the mean score for HYDRA-generated peptides. \textbf{(C)} represents the pairwise Tanimoto diversity scores across the generated peptide sets for each binding target. While the average diversity scores are similar, the distribution of Tanimoto scores implies greater consistency in HYDRA's peptide diversity compared to RFDiffusion.}

  \label{fig:performance}
\end{figure}

\begin{figure}
  \center

  \psfig{figure=figures/plots/gen_vina_top1.pdf,width=0.45\linewidth} \\
  \psfig{figure=figures/plots/gen_vina_mean.pdf,width=0.45\linewidth}
  \psfig{figure=figures/plots/gen_vina_median.pdf,width=0.45\linewidth} \\
  \psfig{figure=figures/plots/gen_vina_common_top1.pdf,width=0.9\linewidth} \\
  \psfig{figure=figures/plots/gen_vina_common_median.pdf,width=0.9\linewidth} \\

  \caption{Comparing the binding affinities of peptides generated by HYDRA and RFDiffusion.}

  \label{fig:extended_vina_metrics}
\end{figure}

\begin{figure}
  \center

  \psfig{figure=figures/plots/gen_frodock_top1.pdf,width=0.45\linewidth} \\
  \psfig{figure=figures/plots/gen_frodock_mean.pdf,width=0.45\linewidth}
  \psfig{figure=figures/plots/gen_frodock_median.pdf,width=0.45\linewidth} \\
  \psfig{figure=figures/plots/gen_frodock_common_top1.pdf,width=0.9\linewidth} \\
  \psfig{figure=figures/plots/gen_frodock_common_median.pdf,width=0.9\linewidth} \\

  \caption{Comparing the FRODOCK scores of peptides generated by HYDRA and RFDiffusion.}

  \label{fig:extended_frodock_metrics}
\end{figure}

In terms of diversity, both HYDRA and RFDiffusion-generated peptides exhibit similar mean Tanimoto scores around 0.6. However, Figure \ref{fig:performance}C reveals key differences in the distribution of scores across targets. HYDRA displays greater consistency in generating peptides with diversity scores close to the mean, suggesting a more uniform level of diversity. However, RFDiffusion exhibits a much wider variance in scores, indicating a less predictable level of diversity across generated peptides. \\

Analysis of physicochemical properties revealed that both models produced peptides within the established range for drug-like molecules in terms of molecular weight. However, HYDRA-designed peptides had isoelectric points closer to 7, indicating potentially greater solubility and stability compared to RFDiffusion-generated ones. Interestingly, the half-life data for both models showed a right-skewed distribution, meaning a few outliers skewed the mean values. While RFDiffusion peptides had a higher average half-life, HYDRA peptides boasted a higher median half-life, suggesting a more consistent level of stability across the generated set. Furthermore, HYDRA demonstrated a clear advantage in peptide stability. The percentage of stable peptides generated by HYDRA was significantly higher (58.31\%) compared to RFDiffusion (31.02\%). This trend was further confirmed by the lower instability index distribution for HYDRA peptides. Aliphatic Index values were comparable for both sets, with HYDRA having a slight edge in mean AI and RFDiffusion edging out on the median value. Finally, the GRAVY scores (generally negative for peptides) indicated higher aqueous solubility for RFDiffusion peptides compared to those generated by HYDRA. \\

The performance gap between HYDRA and RFDiffusion likely stems from their underlying design principles. HYDRA incorporates an explicit binding affinity optimization step, directing the generated peptides toward forming more stable complexes with target proteins. Conversely, RFDiffusion operates in a purely data-driven manner, resulting in peptides directly reflecting the distribution of the training data.
A detailed comparison of both methods across various metrics is presented in Tables \ref{tab:metrics} and \ref{tab:pc_metrics}. Figures \ref{fig:performance}C and \ref{fig:pc_performance}C depict the distribution plots for the aforementioned properties.

\begin{table}[ht]
\centering
\renewcommand{\arraystretch}{1.5}
\resizebox{\textwidth}{!}{%
\begin{tabular}{|c|cc|cc|cc|c|}
      \hline
      & \multicolumn{2}{c|}{Binding Affinity ($\downarrow$)} & \multicolumn{2}{c|}{FRODOCK Score ($\uparrow$)} & \multicolumn{2}{c|}{Diversity ($\uparrow$)} & Stability ($\uparrow$) \\
       Model & Mean $\pm$ SD & Med. & Mean $\pm$ SD & Med. & Mean $\pm$ SD & Med. & \\
      \hline
      HYDRA          & \textbf{-4.112 $\pm$ 1.097} & \textbf{-3.920} & \textbf{2277.171 $\pm$ 
      517.220} & \textbf{2217.161} & \textbf{0.625 $\pm$ 0.066} & \textbf{0.623} & \textbf{58.31\%} \\
      \hline
      RFDiffusion    & -2.764 $\pm$ 0.951 & -2.745 & 1728.622 $\pm$ 642.427 & 1772.440 & 0.579 $\pm$ 0.327 & 0.600 & 31.02\% \\
      \hline
\end{tabular}%
}
\renewcommand{\arraystretch}{1}
\caption{Comparison of metrics for \textit{de novo} peptides generated by HYDRA and RFDiffusion.}
\label{tab:metrics}
\end{table}

\begin{table}[ht]
\centering
\renewcommand{\arraystretch}{1.5}
\resizebox{\textwidth}{!}{%
\begin{tabular}{|c|cc|cc|cc|cc|cc|cc|cc|}
      \hline
      & \multicolumn{2}{c|}{MW [Daltons]} & \multicolumn{2}{c|}{pI} & \multicolumn{2}{c|}{$t_{1/2}$ [hours] ($\uparrow$)} & \multicolumn{2}{c|}{II ($\downarrow$)} & \multicolumn{2}{c|}{AI ($\uparrow$)} & \multicolumn{2}{c|}{GRAVY} \\
       Model & Mean & Med. & Mean & Med. & Mean & Med. & Mean & Med. & Mean & Med. & Mean & Med. \\
      \hline
      HYDRA          & 1109.64 & 1099.28 & \textbf{6.97} & \textbf{6.00} & 10.89 & \textbf{3.50} & \textbf{41.35} & \textbf{35.09} & \textbf{90.39} & 80.00 & -0.20 & -0.21 \\
      \hline
      RFDiffusion    & 1326.58 & 1258.94 & 6.07 & 5.49 & \textbf{17.65} & 1.90 & 84.88 & 73.58 & 88.55 & \textbf{86.67} & -0.93 & -0.86 \\
      \hline
\end{tabular}%
}
\renewcommand{\arraystretch}{1}
\caption{Comparison of different physicochemical properties of peptides generated by HYDRA and RFdiffusion.}
\label{tab:pc_metrics}
\end{table}

\begin{figure}
  \center

  \psfig{figure=figures/plots/gen_pc_mw.pdf,width=0.45\linewidth}
  \psfig{figure=figures/plots/gen_pc_ii.pdf,width=0.45\linewidth}
  \psfig{figure=figures/plots/gen_pc_ai.pdf,width=0.45\linewidth}
  \psfig{figure=figures/plots/gen_pc_pi.pdf,width=0.45\linewidth}
  \psfig{figure=figures/plots/gen_pc_hl.pdf,width=0.45\linewidth}
  \psfig{figure=figures/plots/gen_pc_gravy.pdf,width=0.45\linewidth}

  \caption{\textbf{Comparison of physicochemical properties of peptides generated by HYDRA and RFDiffusion.} Each histogram represents the distribution of peptides across different ranges for a specific property. The y-axis indicates the percentage of peptides within each range. \textbf{(A)} Molecular Weight in Daltons, \textbf{(B)} Instability Index, \textbf{(C)} Aliphatic Index, \textbf{(D)} Isoelectric Point, \textbf{(E)} Half-life in hours, \textbf{(F)} GRAVY Index. Peptides designed by HYDRA show a more favorable bias in their property distribution compared to RFDiffusion.}

  \label{fig:pc_performance}
\end{figure}

\section{Case Study 2: PfEMP1 Protein Targets}
Our single-cell RNA sequencing (scRNA-seq) data revealed that certain PfEMP1 variants are highly expressed. We have utilized these highly expressed PfEMP1 proteins as targets, as they can potentially inhibit the binding of RBCs to blood vessels. \\

The emergence of parasite resistance to established antimalarial drugs like chloroquine and sulfadoxine-pyrimethamine poses a critical challenge. This resistance hampers treatment efficacy, potentially leading to prolonged illness, increased healthcare costs, and elevated mortality risks. Beyond immediate mortality, malaria can have lasting detrimental effects on individuals, even in non-fatal cases. Recurrent infections can contribute to anemia, cognitive decline (particularly in children), and other complications, ultimately diminishing quality of life and economic productivity \cite{shukla2023supervised}. Drug resistance presents a central obstacle in malaria control. The effectiveness of new drugs wanes as the Plasmodium parasite develops resistance mechanisms. \\

In an effort to design \textit{de novo} therapeutic peptide binders against these parasites, we analyzed single-cell transcriptomics data of \textit{P. falciparum} clone 3D7 to identify highly expressed PfEMP1s (PF3D7\_1200600, PF3D7\_1150400, PF3D7\_0712400) based on cellular heterogeneity, cell-to-cell variability, and cellular states \cite{choudhuri2024computational}. Using CAVITY \cite{Yuan2013}, we predicted strong and medium binding sites based on their druggability score. Combining data from CAVITY and Protein Data Bank (PDB) \cite{Berman2000}, we identified extracellular domain binding sites crucial for potential drug interaction. After identifying binding sites, we used HYDRA to generate a set of candidate therapeutic peptides. ExPASy ProtParam \cite{gasteiger2003expasy} then assessed their properties (molecular weight, isoelectric point, half-life, stability, and hydrophobicity). For batch processing of peptides, an open-source Python command-line program wrapping around the ExPASy ProtParam Webserver was built and is available for free use at \texttt{https://github.com/v15hv4/easyprotparam}. We prioritized stable peptides with high predicted binding affinities and favorable biophysical properties. Finally, to gain a complete understanding of their potential as therapeutic agents, we selected a few peptides per protein for further analysis. \\

Through this rigorous selection process, we identified a final set of 12 candidate peptides as presented in Table \ref{tab:drugs}. These peptides are promising therapeutic agents targeting 3 distinct PfEMP1 variants. Furthermore, to assess the potential for broad-spectrum activity, we evaluated the binding affinity of these peptides with additional target receptors beyond those initially considered during their design, the results of which are illustrated in Figure \ref{fig:multi_binding_b}. This analysis aimed to determine if any individual peptide could potentially bind to and interact with binding sites on multiple proteins, offering a broader therapeutic scope. We observe that two of the generated peptides, GPMSAGAGATGM and RLKAVIP, exhibit comparable binding affinities towards pockets 1150400 and 0712400, signaling a potential interaction with both sites. \\

\begin{table}[htbp]
\centering
\renewcommand{\arraystretch}{1.5}
\begin{tabular}{|l|l|p{6cm}|}
    \hline
    Target Protein & Binding Site Domain & Generated Peptide Sequences  \\
    \hline
    PF3D7\_1200600 & Extracellular & \makecell[l]{\\ SLGEVGAPALNFDA \\ ADPTPAHKEGS \\ LVVNTFVAGG \\ MDLTNPGANP \\ GPKTGGNGKSG \\ \hfill}  \\
    \hline
    PF3D7\_1150400 & Extracellular & \makecell[l]{\\ GPMSAGAGATGM \\ NNGAGGDKTV \\ LGSTGMMVNP \\ GGSKFTGASTS \\ \hfill} \\
    \hline
    PF3D7\_0712400 & Extracellular & \makecell[l]{\\ FSVYGIVH \\ RLKAVIP \\ IVVGPAYG \\ \hfill} \\
    \hline
\end{tabular}
\renewcommand{\arraystretch}{1}
\caption{\label{tab:drugs} Identified binding site domains and target peptides selected following the filtering process.}
\end{table}

Moving forward, the next crucial step in this research involves sending these selected peptides for wet lab experiments. This practical evaluation will be instrumental in validating our \textit{in silico} findings and assessing the efficacy and safety of these peptides in a biological context. \\
