\section{Dataset for Training and Evaluation}
To construct training and evaluation sets, we curated peptide structures from the PepBDB database \cite{Wen2019}. PepBDB (Peptide Binding DataBase) is a curated structural database specializing in biological peptide-protein interactions \cite{Wen2019}. It provides clean data for structure-based peptide drug design, particularly for docking and scoring studies. Compiled from the Protein Data Bank (PDB), PepBDB focuses on structures of interacting peptide-protein complexes, with peptides limited to 50 amino acid residues in length. Regular monthly updates ensure the database reflects the latest data released in the PDB.
We curated 9225 protein-peptide complexes for the training dataset, 200 complexes for the validation dataset as well as 193 complexes for the test dataset. The complete dataset comprising 9618 protein-peptide complexes contained 1082 complexes that shared receptors with a subset of 504 unique receptors. To minimize overfitting and training set bias, the test set was carefully selected to have minimal overlap with the receptor proteins present in the train set. This resulted in the test set of 193 complexes with only 7 complexes sharing receptor proteins found in the training set. This minimal overlap between the training and test set data assures that our evaluation metric values are not a product of overfitting or train set bias. Table TODO summarizes the analysis of various physicochemical properties of peptides found in the dataset. A detailed description of the data curation methods is provided in the Supplementary Material.
To assess HYDRA's ability to generalize to unseen receptors, we focused on \textit{de novo} peptide binder generation (novel peptides not encountered during training) for each of the 193 binding pockets within the independent test set. For each pocket, we generated 30 unique peptides, resulting in a total of 5790 novel peptides to evaluate HYDRA's binding prediction for unseen receptors.

\section{Model} % TODO: elaborate on this a bit
The SE(3)-equivariant network contains 9 equivariant layers where $f_h$ and $f_x$ are implemented as graph attention layers for features and coordinates with 16 attention heads and 128 hidden features. For Binary PSO, 50 particles are instantiated with cognitive (c1) and social (c2) acceleration coefficients set to 2.5 and 0.5, respectively. Additionally, the inertia weight is set to 0.9.
The determination of these parameters is justified in the Supplementary Materials.

\section{Training}
The model was trained using the Adam optimizer \cite{kingma2014adam}, with an initial learning rate of $10^{-3}$. To prevent overfitting and improve generalization, a data augmentation strategy was employed during training. This involved adding a small Gaussian noise with a standard deviation of 0.1 to the protein atom coordinates. The forward and reverse diffusion processes took place through 1000 steps. Additionally, a learning rate decay schedule was implemented to decay the learning rate exponentially with a factor of 0.6 towards a minimum value of $10^{-6}$ if there is no improvement in the validation loss for 10 consecutive steps. The model was trained using a batch size of 2, and to balance the contributions of different loss terms within the overall loss function, a factor of $\alpha = 100$ was multiplied onto the residue type loss. Figures concerning the training progression are provided in the Supplementary Materials.

\vspace{3pt} \noindent
The deep diffusion model was trained on 4x NVIDIA GeForce RTX 2080 Ti GPUs using the Distributed Data Parallel (DDP) Strategy. All inference and reconstruction experiments were carried out on multiple nodes, each with 40x Intel Xeon E5-2640 v4 CPUs, 80 GB of RAM, and 1x NVIDIA GeForce RTX 2080 Ti GPU.

\section{Peptide generation for PfEMP1 proteins}
\textit{Plasmodium falciparum}, the causative agent of the most severe form of malaria, expresses a diverse family of PfEMP1 proteins. These highly polymorphic surface antigens, comprising approximately 60 known variants, play a critical role in severe malaria pathogenesis by mediating the cytoadherence of infected erythrocytes to endothelial receptors. This adherence leads to sequestration and contributes to tissue damage \cite{jensen2020cerebral}. PfEMP1 is a pivotal virulence factor secreted by the malaria parasite. It binds to the erythrocyte membrane, triggering the binding of red blood cells (RBCs) to blood vessels \cite{pasternak2009pfemp1}. By obstructing tiny blood arteries, they exacerbate malaria infections and increase the risk of cerebral malaria, placental malaria, and severe anemia \cite{jensen2020cerebral}. The parasite can avoid the host immune system because of this antigenic diversity of PfEMP1 family genes, since new infection can express different PfEMP1 variants that are not recognized by preexisting immune responses \cite{scherf2008antigenic}.
We leveraged single-cell RNA-seq data to identify the five most highly expressed PfEMP1 genes. Using CAVITY \cite{Yuan2013}, we predicted strong and medium druggable binding sites within the proteins encoded by these genes (Figure TODO). Subsequently, HYDRA was used to design potential peptide molecules specifically targeted to these binding sites (Figure TODO). We also checked the binding affinities of generated peptides of one with other proteins (heatmap TODO).  During analysis of the protein structures, we focused on extracellular and intracellular domains for peptide generation, leveraging cavities as potential binding pockets. Finally, we evaluated the binding affinities between the designed peptides and their target proteins on these PfEMP1 variants.
